wearos/src/main/java/com/habitrpg/wearos/habitica/ui/viewmodels/TaskListViewModel.kt

```diff
@@ -65,4 +64,90 @@ class TaskListViewModel @Inject constructor(
             onResult(result)
         }
     }
+
+    private fun map(tasks: List<Task>): List<Task> {
+        taskCount.value = tasks.size
+        return tasks
+    }
+
+    private fun mapDaily(tasks: List<Task>): MutableList<Any> {
+        val taskList: MutableList<Any> = tasks.filter { it.isDue == true || it.type == TaskType.TODO }.sortedBy { it.completed }.toMutableList()
+        val firstCompletedIndex = taskList.indexOfFirst { it is Task &&  it.completed }
+        if (firstCompletedIndex >= 0) {
+            // since this is the index of the first completed task, this is also the number of incomplete tasks
+            taskCount.value = firstCompletedIndex
+            taskList.add(firstCompletedIndex, "Done today")
+        } else {
+            taskCount.value = taskList.size
+        }
+        return taskList
+    }
+
+    private fun getCurrentToDos(): List<Any>? {
+        val gson = Gson()
+        val data = mutableListOf<Any>()
+        val tasksString = sharedPreferences.getString("to_do_tasks", null)
+        if (tasksString != null) {
+            val type: Type = object : TypeToken<ArrayList<Task?>?>() {}.type
+            val savedCurrentTasks = gson.fromJson(tasksString, type) as MutableList<Task>
+            val list = savedCurrentTasks.sortedBy { it.completed }
+            val firstCompletedIndex = list.indexOfFirst { it.completed }
+            return if (firstCompletedIndex >= 0) {
+                // since this is the index of the first completed task, this is also the number of incomplete tasks
+                taskCount.value = firstCompletedIndex
+                data.addAll(list)
+                data.add(firstCompletedIndex, "Done today")
+                data
+            } else {
+                savedCurrentTasks
+            }
+        }
+        return null
+    }
+
+    private fun mapTodos(tasks: List<Task>): List<Any>? {
+        saveCurrentToDos(tasks)
+        return getCurrentToDos()
+    }
+
+    private fun saveCurrentToDos(tasks: List<Task>) {
+        val taskList = mutableListOf<Task>()
+        val type: Type = object : TypeToken<ArrayList<Task?>?>() {}.type
+        if (tasksString != null) {
+            val savedCurrentTasks = gson.fromJson(tasksString, type) as MutableList<Task>
+            if (!savedCurrentTasks.isNullOrEmpty()) {
+                for (task in tasks) {
+                    if (!savedCurrentTasks.contains(task)) {
+                        taskList.add(task)
```

Comment:
I would only save completed todos and not all todos, since for incomplete todos we use the the repository as data source.